cmd/integration-test-server/main.go

```diff
@@ -0,0 +1,195 @@
+					shardState.SyncState.FillEmptyProgress < oldShardState.SyncState.FillEmptyProgress) {
+				addErrorMessage(fmt.Sprintf("es-node fill empty progress do not increase in %f minutes, "+
+					"old filled: %d, new filled %d; old progress: %d, new progress: %d", expectedStateRefreshTime.Minutes(), oldShardState.SyncState.EmptyFilled,
+					shardState.SyncState.EmptyFilled, oldShardState.SyncState.FillEmptyProgress, shardState.SyncState.FillEmptyProgress))
+			}
+
+			if oldShardState.SyncState.FillEmptyProgress == 10000 && oldShardState.SyncState.SyncProgress == 10000 &&
+				(shardState.MiningState.MiningPower == 0 || shardState.MiningState.SamplingTime == 0) {
+				addErrorMessage("Mining should be start after sync done.")
+			}
+		}
+		if !check {
+			addErrorMessage(fmt.Sprintf("Shard %d in the new state do not exist in the old state", shardState.ShardId))
+		}
+	}
+}
+
+func checkFinalState(state *node.NodeState) {
+	if state == nil {
+		addErrorMessage("No state submitted during the test")
+		return
+	}
+	if !hasConnectedPeer {
+		addErrorMessage("es-node peer count should larger than 0")
+	}
+
+	log.Info("Final state", "id", state.Id, "version", state.Version)
+	for _, shardState := range state.Shards {
+		if shardState.SyncState.SyncProgress != 10000 {
+			addErrorMessage("Sync should be finished during the test")
+		}
+		if shardState.SyncState.FillEmptyProgress != 10000 {
+			addErrorMessage("Fill should be finished during the test")
+		}
+		if shardState.MiningState.SamplingTime == 0 || shardState.MiningState.MiningPower == 0 {
+			addErrorMessage("Mining should be start after sync done.")
+		}
+		if shardState.SubmissionState.LastSucceededTime == 0 || shardState.SubmissionState.Succeeded == 0 {
+			addErrorMessage("At lease one block should be mined successfully during the test.")
+		}
+		if shardState.SubmissionState.Failed > 0 {
+			addErrorMessage(fmt.Sprintf("%d submission failed during the test.", shardState.SubmissionState.Failed))
+		}
+		if shardState.SubmissionState.Dropped > 0 {
+			addErrorMessage(fmt.Sprintf("%d submission dropped during the test.", shardState.SubmissionState.Dropped))
+		}
+		log.Info("Final state", "id", state.Id, "shard", shardState.ShardId, "miner", shardState.Miner, "sync progress",
+			shardState.SyncState.SyncProgress, "fill progress", shardState.SyncState.FillEmptyProgress, "mining power",
+			shardState.MiningState.MiningPower, "sampling time", shardState.MiningState.SamplingTime, "succeeded submission",
+			shardState.SubmissionState.Succeeded, "failed submission", shardState.SubmissionState.Failed, "dropped submission",
+			shardState.SubmissionState.Dropped, "last succeeded time", shardState.SubmissionState.LastSucceededTime)
+	}
+}
+
+func addErrorMessage(errMessage string) {
+	log.Warn("Add error message", "msg", errMessage)
+	errorMessages = append(errorMessages, errMessage+"\n")
+}
+
+func listenAndServe(port int) error {
+	http.HandleFunc("/hello", HelloHandler)
```

Comment:
Who calls the 2 RPC methods?

Reply:
es-node will call this to register to the metrics service/integration server.
![1723774006184](https://github.com/user-attachments/assets/0d50d6c2-8596-4b48-b3d5-f116ac8717fe)